lib/membrane/rtp/demuxer/jitter_buffer.ex

```diff
@@ -0,0 +1,233 @@
+    %State{jitter_buffer_state | max_latency_timer: nil}
+  end
+
+  @spec initial_latency_passed(State.t()) :: State.t()
+  def initial_latency_passed(jitter_buffer_state) do
+    %State{jitter_buffer_state | initial_latency_waiting: false}
+  end
+
+  @spec insert_buffer(State.t(), Membrane.Buffer.t(), Demuxer.stream_phase()) :: State.t()
+  def insert_buffer(jitter_buffer_state, buffer, stream_phase) do
+    case stream_phase do
+      :timed_out ->
+        jitter_buffer_state
+
+      phase when phase in [:waiting_for_matching_pad, :matched_with_pad] ->
+        case BufferStore.insert_buffer(jitter_buffer_state.buffer_store, buffer) do
+          {:ok, buffer_store} ->
+            %State{jitter_buffer_state | buffer_store: buffer_store}
+
+          {:error, :late_packet} ->
+            Membrane.Logger.debug("Late packet has arrived")
+            jitter_buffer_state
+        end
+    end
+  end
+
+  @spec get_output_actions(State.t(), Demuxer.stream_phase()) ::
+          {[Membrane.Element.Action.t()], State.t()}
+  def get_output_actions(jitter_buffer_state, stream_phase) do
+    if stream_phase == :matched_with_pad and not jitter_buffer_state.initial_latency_waiting do
+      {too_old_records, buffer_store} =
+        BufferStore.flush_older_than(
+          jitter_buffer_state.buffer_store,
+          jitter_buffer_state.latency
+        )
+
+      {buffers, buffer_store} = BufferStore.flush_ordered(buffer_store)
+      jitter_buffer_state = %State{jitter_buffer_state | buffer_store: buffer_store}
+
+      {actions, jitter_buffer_state} =
+        (too_old_records ++ buffers)
+        |> Enum.flat_map_reduce(jitter_buffer_state, &record_to_action/2)
+
+      jitter_buffer_state = set_timer(jitter_buffer_state)
+
+      {actions, jitter_buffer_state}
+    else
+      {[], jitter_buffer_state}
+    end
+  end
+
+  @spec get_end_of_stream_actions(State.t()) :: [Membrane.Element.Action.t()]
+  def get_end_of_stream_actions(jitter_buffer_state) do
+    if jitter_buffer_state.pad == nil do
+      []
+    else
+      {actions, _jitter_buffer_state} =
+        jitter_buffer_state.buffer_store
+        |> BufferStore.dump()
+        |> Enum.flat_map_reduce(jitter_buffer_state, &record_to_action/2)
+
+      actions ++ [end_of_stream: jitter_buffer_state.pad]
+    end
+  end
+
+  @spec set_timer(State.t()) :: State.t()
+  defp set_timer(%State{max_latency_timer: nil, latency: latency} = jitter_buffer_state)
+       when latency > 0 do
+    new_timer =
+      case BufferStore.first_record_timestamp(jitter_buffer_state.buffer_store) do
+        nil ->
+          nil
+
+        buffer_ts ->
+          since_insertion = Time.monotonic_time() - buffer_ts
+          send_after_time = Time.as_milliseconds(latency - since_insertion, :round)
+
+          if send_after_time > 0 do
+            Process.send_after(self(), {:send_actions, jitter_buffer_state.ssrc}, send_after_time)
+          else
+            nil
+          end
```

Comment:
[NIT]
```suggestion
          if send_after_time > 0, do: Process.send_after(self(), {:send_actions, jitter_buffer_state.ssrc}, send_after_time)
```

Reply:
It feels slightly bloaty and that is why I thought of writing it in a more concise way, but sure, do what you are comfortable with doing ;)